coldfront/core/utils/management/commands/audit_data.py

```diff
@@ -0,0 +1,323 @@
+            - Inactive ones (ones that were not renewed) should have a start
+                date set to the start of the current allowance year
+                AllocationPeriod, but no end date.
+            - Active ones (ones that were renewed, or were created during
+                this allowance year) should have a start before and an end date
+                equal to that of the current allowance year AllocationPeriod.
+          ICAs (Instructional Computing Allowances) are valid within a
+          particular UC Berkeley semester.
+            - Inactive ones should have a start date, but no end date.
+            - Active ones (currently none on production) should match a current
+                (has started, but has not ended) instructional AllocationPeriod.
+          Recharge and Condo may/should have start dates, but don’t have end
+          dates, as they don’t end.
+        '''
+        allocations = Allocation.objects.select_related('project',
+                            'project__status').prefetch_related('resources') \
+            .filter(resources__name__endswith='Compute') \
+            .order_by('resources__name', 'project__name',
+                      'project__status__name', 'start_date', 'end_date') \
+            .values('id', 'project__status__name',
+                   'project__name','start_date', 'end_date',
+                   'resources__name')
+
+        FCA_PCA_ALLOCATION_PERIOD = get_current_allowance_year_period()
+        ICA_ALLOCATION_PERIODS = AllocationPeriod.objects.filter(
+                                    Q(end_date__gt=display_time_zone_current_date())
+                                    & (Q(name__startswith='Fall Semester')
+                                       | Q(name__startswith='Spring Semester')
+                                       | Q(name__startswith='Summer Sessions') \
+                                      )).all()
+
+        # TODO: Change these if lrc_enabled
+        FCA_PCA_PREFIXES = ('fc_', 'pc_')
+        ICA_PREFIX = 'ic_'
+        RECHARGE_CONDO_PREFIX = ('co_', 'ac_')
+
+        for allocation in allocations:
+            id = allocation['id']
+            start_date = allocation['start_date']
+            end_date = allocation['end_date']
+            project_status = allocation['project__status__name']
+            project = allocation['project__name']
+            resource = allocation['resources__name']
+
+            if end_date is not None and end_date < start_date:
+                self.stdout.write(self.style.ERROR(f'{resource} {id} '
+                f'for {project_status.lower()} project {project} has an end '
+                f'date before its start date.'))
+
+            if project.startswith(FCA_PCA_PREFIXES):
+                if project_status == 'Inactive' \
+                        and start_date != FCA_PCA_ALLOCATION_PERIOD.start_date:
+                    self.stdout.write(self.style.ERROR(f'{resource} {id} '
```

Comment:
Neither deployment conforms to this check at the moment, for different reasons.

On LRC, we ran the allocation renewals a bit late this year to coincide with launch. Let's apply the `<` check from L156 to `Inactive` projects as well.

On BRC, I may just end up rewriting start/end dates for inactive projects to be the correct ones.

Reply:
I swapped the != for < if that's what you mean

Reply:
I meant that it's acceptable for both Active and Inactive projects to have start dates greater than or equal to the start date of the corresponding allocation period.